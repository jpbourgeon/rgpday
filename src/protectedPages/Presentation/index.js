import React from 'react'
import loadable from '@loadable/component'
import { Redirect } from '@reach/router'
import { withStyles } from '@material-ui/core/styles'

const styles = () => ({
  '@global': {
    '.grecaptcha-badge': {
      display: 'none'
    }
  }
})

const AsyncPresentation = loadable(props => (import(`../../presentations/${props.presentationId}`)))

const PresentationPage = (props) => {
  const { location, config } = props
  if (config && config.presentationId) {
    return (
      <AsyncPresentation presentationId={config.presentationId} target={location.hash} />
    )
  } else {
    return (<Redirect to='/dashboard' noThrow />)
  }
}

export default withStyles(styles)(PresentationPage)
