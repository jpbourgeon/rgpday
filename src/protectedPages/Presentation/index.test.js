/* global describe, it, expect */
import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import Presentation from './'
import jssSerializer from 'src/utils/jssSnapshotSerializer'

expect.addSnapshotSerializer(jssSerializer)

describe('The 2019-INGE-01 Presentation component', () => {
  it('should render correctly', () => {
    const renderer = new ShallowRenderer()
    renderer.render(<Presentation
      config={{ scenarioId: '2019-INGE-01' }}
      location={{ hash: '#start' }}
    />)
    const tree = renderer.getRenderOutput()
    expect(tree).toMatchSnapshot()
  })
})
