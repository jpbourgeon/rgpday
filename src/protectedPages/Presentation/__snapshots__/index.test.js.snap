// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The 2019-INGE-01 Presentation component should render correctly 1`] = `
<PresentationPage
  classes={Object {}}
  config={
    Object {
      "scenarioId": "2019-INGE-01",
    }
  }
  location={
    Object {
      "hash": "#start",
    }
  }
/>
`;
